packages/tofu/src/inspectSource.js

```diff
@@ -133,44 +133,38 @@ function inspectGlobals(
   }
 }
 
-function inspectEsmImports(ast) {
-  const esmImports = []
+/**
+ * This finds all modules `import`ed into the AST, as well as any re-exported modules.
+ *
+ * @param {import('@babel/types').Node} ast
+ * @param {string[]} [packagesToInspect]
+ * @returns {{ esmImports: string[] }}
+ */
+function inspectEsmImports(ast, packagesToInspect) {
+  const pkgsToInspect = new Set(packagesToInspect)
+  /** @type {Set<string>} */
+  const esmImports = new Set()
+
+  const handleNodePath = (path) => {
+    const importSource = path.node.source?.value
+
+    if (
+      importSource &&
+      ((packagesToInspect && pkgsToInspect.has(importSource)) ||
+        !packagesToInspect)
```

Comment:
Nit: I feel the added parentheses in the second clause don't help and this rewrite makes it more obvious. WDYT?
```suggestion
      !packagesToInspect
      || pkgsToInspect.has(importSource) && importSource
```

Reply:
prettier does this. it does it because not everyone always remembers order of operations

Reply:
This is confusing to me for other reasons. Since we're turning packagesToInspect into the pkgsToInspect set, could we limit ourselves to using just one? I'd prefer if packagesToInspect were not uttered below the first line of the function. 

And since the empty set is not falsy, but doesn't contain anything, `packagesToInspect && pkgsToInspect.has(importSource)) ||       !packagesToInspect` becomes `pkgsToInspect.has(importSource)) || pkgsToInspect.size===0`. Amirite?